spec-dtslint/operators/race-spec.ts

```diff
@@ -0,0 +1,40 @@
+import { of } from 'rxjs';
+import { race } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const o = of('a', 'b', 'c').pipe(race()); // $ExpectType Observable<string>
+});
+
+it('should allow observables', () => {
+  const o = of('a', 'b', 'c').pipe(race(of('x', 'y', 'z'))); // $ExpectType Observable<string>
+  const p = of('a', 'b', 'c').pipe(race(of('x', 'y', 'z'), of('t', 'i', 'm'))); // $ExpectType Observable<string>
+});
+
+it('should allow an array of observables', () => {
+  const o = of('a', 'b', 'c').pipe(race([of('x', 'y', 'z')])); // $ExpectType Observable<string>
+  const p = of('a', 'b', 'c').pipe(race([of('x', 'y', 'z'), of('t', 'i', 'm')])); // $ExpectType Observable<string>
+});
+
+it('should allow observables and arrays', () => {
+  const o = of('a', 'b', 'c').pipe(race(of('x', 'y', 'z'), [of('t', 'i', 'm')])); // $ExpectType Observable<string>
```

Comment:
That's an error.   `race(of('test'), [of('whatever')])` should result in `Observable<Observable<"whatever">>`

Reply:
Okay... weirdly, because sync, this is always correct... but I still think it's a typings error.


of('test').pipe(race(anything, you, want, here))` will always result in `Observable<"test">`. Because the leading observable, `of('test')` will emit synchronously before any of the other sources are subscribed to.

However, I'd expect  `Observable<string>.pipe(race(Observable<string>, Array<Observable<string>>))` to result in `Observable<string|Observable<string>>` as a matter of typings.

Reply:
@timdeschryver can we remove this type check and file an issue? I don't think we want to enforce this.

Reply:
We sure can, but wouldn't it be helpful to have this typing while resolving the issue?(Because you would have an example).